src/improvements/tasks/MultisigTask.sol

```diff
@@ -595,133 +604,145 @@ abstract contract MultisigTask is Test, Script, StateOverrideManager {
         console.log("Child multisig: %s", getAddressLabel(childMultisig));
-
-        bytes memory encodedTxData = getEncodedTransactionData(childMultisig, callData);
-        require(encodedTxData.length == 66, "MultisigTask: encodedTxData length is not 66 bytes.");
-
-        bytes32 safeTxHash;
-        assembly {
-            // 66 bytes = (bytes1(0x19), bytes1(0x01), bytes32(domainSeparator()), bytes32(safeTxHash))
-            // Retrieve the last 32 bytes of encodedTxData (safeTxHash).
-            // Memory layout of encodedTxData:
-            // - The first 32 bytes store the length (66 bytes in this case).
-            // - The actual data starts at encodedTxData + 32.
-            // - The last 32 bytes of the data (safeTxHash) start at:
-            //   encodedTxData + 32 + (66 - 32) = encodedTxData + 66.
-            safeTxHash := mload(add(encodedTxData, mload(encodedTxData)))
-        }
-
-        bytes32 domainSeparator = GnosisSafeHashes.calculateDomainSeparator(block.chainid, childMultisig);
         console.log("Domain Hash:    ", vm.toString(domainSeparator));
-        console.log("Message Hash:   ", vm.toString(safeTxHash));
+        console.log("Message Hash:   ", vm.toString(messageHash));
     }
 
     /// @notice print the tenderly simulation payload with the state overrides
-    function printTenderlySimulationData(Action[] memory actions) internal view {
+    function printTenderlySimulationData(Action[] memory actions, address optionalChildMultisig) internal view {
+        // TODO: Support child nonce as a state override. Right now we always get the latest nonce.
+        // Use the max uint256 to indicate that the child multisig nonce is not provided (zero is a valid nonce).
+        uint256 childMultisigNonce =
+            optionalChildMultisig != address(0) ? _getNonce(optionalChildMultisig) : type(uint256).max;
         Simulation.StateOverride[] memory allStateOverrides =
-            getStateOverrides(parentMultisig, _getNonce(parentMultisig));
+            getStateOverrides(parentMultisig, _getNonce(parentMultisig), optionalChildMultisig, childMultisigNonce);
 
-        bytes memory txData = _execTransationCalldata(
-            parentMultisig, getMulticall3Calldata(actions), Signatures.genPrevalidatedSignature(msg.sender)
-        );
+        if (optionalChildMultisig != address(0)) {
+            bytes memory finalExec = getNestedSimulationMulticall3Calldata(actions, optionalChildMultisig);
+
+            logTenderlySimulationPayload(finalExec, allStateOverrides, MULTICALL3_ADDRESS);
+
+            console.log("\nSimulation link:");
+            Simulation.logSimulationLink({
+                _to: MULTICALL3_ADDRESS,
+                _data: finalExec,
+                _from: msg.sender,
+                _overrides: allStateOverrides
+            });
+            // TODO: Simulate the new Tenderly calldata locally (nested and non-nested).
```

Comment:
We don't need to simulate the tenderly calldata locally. We did that before because we didn't have what we did have now.

Reply:
Remove comment.

Reply:
Done